File: Leetcode/Python/3174. Clear Digits/solution.py
# https://leetcode.com/problems/clear-digits/description

class Solution:
    def clearDigits(self, s: str) -> str:
        char_stack = []

        for char in s:
            if not char.isdigit():
                char_stack.append(char)
            else:
                char_stack.pop()

        return "".join(char_stack)

File: Leetcode/Typescript/3174. Clear Digits/solution.ts
// https://leetcode.com/problems/clear-digits/description

function clearDigits(s: string): string {
  const strStack: string[] = [];

  for (let idx = 0; idx < s.length; idx++) {
    if (isNaN(s[idx] as any)) {
      strStack.push(s[idx]);
    } else {
      strStack.pop();
    }
  }
  return strStack.join("");
}
